src/Dfc.ProviderPortal.TribalExporter/Functions/Export.cs

```diff
@@ -56,14 +58,28 @@ public static class Export
 
             try
             {
+                log.LogInformation("[Export] grabbing providers");
                 var providersForExport = await GetProvidersFromCsv(migrationProviderCsv, blobStorageHelper, logFile, containerProviderFiles);
 
-                await GenerateProvidersExport(providerCollectionService, providersForExport, logFile, providersFileName, containerExporter, containerNameExporter, fileNames);
+                var providerFileName = await GenerateProvidersExport(providerCollectionService, providersForExport, logFile, providersFileName, containerExporter, containerNameExporter);
+                fileNames.Add(providerFileName);
+
+                var count = 0;
+                var total = providersForExport.Count();
 
-                foreach (var provider in providersForExport)
+                Parallel.ForEach(providersForExport, async (provider) =>
```

Comment:
Is `Parallel.ForEach` correct here? Not sure each delegate is `await`ed so may be a race..?

Reply:
Maybe `Task.WhenAll(providersForExport.Select(async provider...`